### ISSUE TITLE
Stitchdata requires `innodb_lock_wait_timeout` be defined to connect to a Dolt database through MySQL connector

### ISSUE THREAD

**Issue by timsehn** [dolthub/dolt#4190]
Pulling this from this uber issue: #4007 

[Stitchdata](https://www.stitchdata.com/) is a tool to pull data out of an OLTP database and insert it into a data warehouse. When used with Dolt through the MySQL connector, Stitchdata successfully connects through tunnel, but hits fatal error on `Error: %s Unknown system variable 'innodb_lock_wait_timeout'`

```
2022-08-03 22:08:53,635Z   main - INFO Running tap-hp-mysql version 1.3.2 and target-stitch-avro version 0.2.2
2022-08-03 22:08:54,808Z   main - INFO [smart-services] building smart service producer: topic( com.stitchdata.extractionJobStarted ), use_ssl( true )
2022-08-03 22:08:54,820Z   main - INFO [smart-services] building smart service producer: topic( com.stitchdata.extractionJobFinished ), use_ssl( true )
2022-08-03 22:08:54,830Z   main - INFO [smart-services] building smart service producer: topic( com.stitchdata.streamRecordCount ), use_ssl( true )
<SSH TUNNEL CONFIG SNIPPED>
2022-08-03 22:08:55,896Z   main - INFO Successfully connected to tunnel
2022-08-03 22:08:57,381Z   main - INFO [smart-services] event successfully sent to kafka: com.stitchdata.extractionJobStarted [16] at offset None
2022-08-03 22:08:57,382Z   main - INFO Starting tap to discover schemas: tap-env/bin/tap-hp-mysql --config /tmp/tap_discover_config.json --discover
2022-08-03 22:09:00,127Z    tap - WARNING: error-handler already refers to: #'clojure.core/error-handler in namespace: com.stitchdata.target-stitch-avro.flush-pipeline, being replaced by: #'com.stitchdata.target-stitch-avro.flush-pipeline/error-handler
2022-08-03 22:09:00,742Z    tap - INFO  [main] tap-hp-mysql.main - Started nrepl server at /127.0.0.1:45453
2022-08-03 22:09:00,764Z    tap - INFO  [main] tap-hp-mysql.core - Starting discovery mode
2022-08-03 22:09:01,253Z    tap - INFO  [main] tap-hp-mysql.config - Connecting to MySQL version 5.7.9-Vitess
2022-08-03 22:09:01,455Z    tap - WARN  [main] tap-hp-mysql.config - Encountered error checking server params. Error: %s Unknown system variable 'innodb_lock_wait_timeout'
2022-08-03 22:09:01,465Z    tap - FATAL [main] tap-hp-mysql.main - Fatal Error Occurred - 
2022-08-03 22:09:01,482Z    tap - ERROR [main] #error {
2022-08-03 22:09:01,482Z    tap -  :cause nil
2022-08-03 22:09:01,482Z    tap -  :via
2022-08-03 22:09:01,482Z    tap -  [{:type java.lang.NullPointerException
2022-08-03 22:09:01,483Z    tap -    :message nil
2022-08-03 22:09:01,483Z    tap -    :at [clojure.string$lower_case invokeStatic string.clj 217]}]
2022-08-03 22:09:01,483Z    tap -  :trace
2022-08-03 22:09:01,483Z    tap -  [[clojure.string$lower_case invokeStatic string.clj 217]
2022-08-03 22:09:01,483Z    tap -   [clojure.string$lower_case invoke string.clj 213]
2022-08-03 22:09:01,483Z    tap -   [tap_hp_mysql.config$maybe_add_mariadb_compatibility invokeStatic config.clj 61]
2022-08-03 22:09:01,483Z    tap -   [tap_hp_mysql.config$maybe_add_mariadb_compatibility invoke config.clj 55]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.config$check_connection invokeStatic config.clj 72]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.config$check_connection invoke config.clj 65]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.config$__GT_conn_map_STAR_ invokeStatic config.clj 121]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.config$__GT_conn_map_STAR_ invoke config.clj 99]
2022-08-03 22:09:01,484Z    tap -   [clojure.lang.AFn applyToHelper AFn.java 154]
2022-08-03 22:09:01,484Z    tap -   [clojure.lang.AFn applyTo AFn.java 144]
2022-08-03 22:09:01,484Z    tap -   [clojure.core$apply invokeStatic core.clj 665]
2022-08-03 22:09:01,484Z    tap -   [clojure.core$memoize$fn__6877 doInvoke core.clj 6353]
2022-08-03 22:09:01,484Z    tap -   [clojure.lang.RestFn invoke RestFn.java 408]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.discovery$build_catalog invokeStatic discovery.clj 173]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.discovery$build_catalog invoke discovery.clj 172]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.discovery$discover invokeStatic discovery.clj 181]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.discovery$discover invoke discovery.clj 179]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.core$do_discovery invokeStatic core.clj 19]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.core$do_discovery invoke core.clj 17]
2022-08-03 22:09:01,484Z    tap -   [tap_hp_mysql.main$main_impl invokeStatic main.clj 98]
2022-08-03 22:09:01,485Z    tap -   [tap_hp_mysql.main$main_impl invoke main.clj 90]
2022-08-03 22:09:01,485Z    tap -   [tap_hp_mysql.main$_main invokeStatic main.clj 147]
2022-08-03 22:09:01,485Z    tap -   [tap_hp_mysql.main$_main doInvoke main.clj 136]
2022-08-03 22:09:01,485Z    tap -   [clojure.lang.RestFn applyTo RestFn.java 137]
2022-08-03 22:09:01,485Z    tap -   [tap_hp_mysql.main main nil -1]]}
2022-08-03 22:09:01,485Z    tap - INFO  [main] com.stitchdata.target-stitch-avro.core - wait-for-target
2022-08-03 22:09:01,485Z    tap - INFO  [main] com.stitchdata.target-stitch-avro.core - flushing remaining pending requests: []
2022-08-03 22:09:01,485Z    tap - INFO  [main] tap-hp-mysql.main - Shutting down the nrepl server
2022-08-03 22:09:01,520Z   main - INFO Tap exited abnormally with status 1
2022-08-03 22:09:02,399Z   main - INFO [smart-services] event successfully sent to kafka: com.stitchdata.extractionJobFinished [27] at offset None
2022-08-03 22:09:02,402Z   main - INFO Closed tunnel and deleted temporary keyfile
2022-08-03 22:09:02,402Z   main - INFO Exit status is: Discovery failed with code 1 and error message: "[main] tap-hp-mysql.main - Fatal Error Occurred - ".
```

**Comment by timsehn** [comment]
This is resolved in 1.16.4

**Comment by macneale4** [comment]
https://github.com/dolthub/dolt/releases/tag/v1.16.4
